plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoConnectorTest.java

```diff
@@ -346,55 +350,75 @@ public void testJson()
+        String tmpArray1 = "tmp_array1" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tmpArray1 + " AS SELECT ARRAY[1, 2, NULL] AS col", 1);
+        assertQuery("SELECT col[2] FROM " + tmpArray1, "SELECT 2");
+        assertQuery("SELECT col[3] FROM " + tmpArray1, "SELECT NULL");
+        assertUpdate("DROP TABLE " + tmpArray1);
+
+        String tmpArray2 = "tmp_array2" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tmpArray2 + " AS SELECT ARRAY[1.0E0, 2.5E0, 3.5E0] AS col", 1);
+        assertQuery("SELECT col[2] FROM " + tmpArray2, "SELECT 2.5");
+        assertUpdate("DROP TABLE " + tmpArray2);
+
+        String tmpArray3 = "tmp_array3" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tmpArray3 + " AS SELECT ARRAY['puppies', 'kittens', NULL] AS col", 1);
+        assertQuery("SELECT col[2] FROM " + tmpArray3, "SELECT 'kittens'");
+        assertQuery("SELECT col[3] FROM " + tmpArray3, "SELECT NULL");
+        assertUpdate("DROP TABLE " + tmpArray3);
+
+        String tmpArray4 = "tmp_array4" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tmpArray4 + " AS SELECT ARRAY[TRUE, NULL] AS col", 1);
+        assertQuery("SELECT col[1] FROM " + tmpArray4, "SELECT TRUE");
+        assertQuery("SELECT col[2] FROM " + tmpArray4, "SELECT NULL");
+        assertUpdate("DROP TABLE " + tmpArray4);
+
+        String tmpArray5 = "tmp_array5" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tmpArray5 + " AS SELECT ARRAY[ARRAY[1, 2], NULL, ARRAY[3, 4]] AS col", 1);
+        assertQuery("SELECT col[1][2] FROM " + tmpArray5, "SELECT 2");
+        assertUpdate("DROP TABLE " + tmpArray5);
+
+        String tmpArray6 = "tmp_array6" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tmpArray6 + " AS SELECT ARRAY[ARRAY['\"hi\"'], NULL, ARRAY['puppies']] AS col", 1);
+        assertQuery("SELECT col[1][1] FROM " + tmpArray6, "SELECT '\"hi\"'");
+        assertQuery("SELECT col[3][1] FROM " + tmpArray6, "SELECT 'puppies'");
+        assertUpdate("DROP TABLE " + tmpArray6);
     }
 
     @Test
     public void testTemporalArrays()
     {
-        assertUpdate("CREATE TABLE tmp_array7 AS SELECT ARRAY[DATE '2014-09-30'] AS col", 1);
-        assertOneNotNullResult("SELECT col[1] FROM tmp_array7");
-        assertUpdate("CREATE TABLE tmp_array8 AS SELECT ARRAY[TIMESTAMP '2001-08-22 03:04:05.321'] AS col", 1);
-        assertOneNotNullResult("SELECT col[1] FROM tmp_array8");
+        String tmpArray7 = "tmp_array7" + randomNameSuffix();
+        String tmpArray8 = "tmp_array8" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tmpArray7 + " AS SELECT ARRAY[DATE '2014-09-30'] AS col", 1);
+        assertOneNotNullResult("SELECT col[1] FROM " + tmpArray7);
+        assertUpdate("DROP TABLE " + tmpArray7);
+        assertUpdate("CREATE TABLE " + tmpArray8 + " AS SELECT ARRAY[TIMESTAMP '2001-08-22 03:04:05.321'] AS col", 1);
```

Comment:
Could you add a new line betweend `DROP` and `CREATE` lines? Also, move `String tmpArray8` to before `CREATE` line.